msal/src/main/java/com/microsoft/identity/client/Account.java

```diff
@@ -97,4 +98,20 @@ String getEnvironment() {
     public Map<String, ?> getClaims() {
         return mIdTokenClaims;
     }
+
+    @NonNull
+    @Override
+    public String getUsername() {
+        String username = "";
+
+        if (null != getClaims()) {
+            final String usernameClaim = (String) getClaims().get(IDToken.PREFERRED_USERNAME);
```

Comment:
Haven't tried it myself, but would this be converted to a string "null"?
https://stackoverflow.com/questions/18723596/no-exception-while-type-casting-with-a-null-in-java

Reply:
might be safer to do null check before casting.

Reply:
It looks like it will still be null 

https://repl.it/repls/FlakyEagerRar

Reply:
Good catch

Reply:
Thanks @shahzaibj for checking - leaving as-is